### ISSUE TITLE
Trying to read a CGYRO file without electrons fails

### ISSUE THREAD

**Issue by d7919** [bpatel2107/pyrokinetics#55]
In `load_local_species` of cgyro there's an assumption that the are electron temperature and density values to normalise to

> ~/.local/lib/python3.8/site-packages/pyrokinetics/cgyro.py in load_local_species(self, pyro, cgyro)
>     327             species_data = local_species[name]
>     328 
> --> 329             species_data.temp = species_data.temp / te
>     330             species_data.dens = species_data.dens / ne
>     331

**Comment by bpatel2107** [comment]
That is the default/recommended settings as laid out here 
https://gafusion.github.io/doc/cgyro.html#quick-links

What scenario are you examining without electrons? If it is adiabatic electrons then we could add a check for that

**Comment by d7919** [comment]
Yes it's an adiabatic electron run.

**Comment by bpatel2107** [comment]
Okay so we'll need to add that in. In general pyrokinetics forces all codes to use the same normalisation where possible. We are considering adding in a few options/letter the user choose their own one, similar to a COCOs. See #33

**Comment by bpatel2107** [comment]
We don't handle adiabatic species in general so it might be worth considering this in a broader sense for each code

**Comment by d7919** [comment]
Yes I think adiabatic/trace/dummy species would be helpful.

I guess here it just needs some default Te/ne/mass, or at least some way to set them. Perhaps can detect no electrons when trying to read things in and add a warning/error unless electron settings are passed.

**Comment by ZedThree** [comment]
Does a method like `LocalSpecies.add_adiabatic_species` make sense? Something like the following:

```python
    def add_adiabatic_species(
        self, name: str, like_species: str, charge: float, mass: float
    ):
        existing_species = self[like_species]
        adiabatic_species = {
            "name": name,
            "mass": mass,
            "z": charge,
            "dens": existing_species.dens,
            "temp": existing_species.temp,
            "vel": existing_species.vel,
            "nu": existing_species.nu,
            "a_lt": existing_species.a_lt,
            "a_ln": existing_species.a_ln,
            "a_lv": existing_species.a_lv,
        }

        self.add_species(name, adiabatic_species)

    def add_adiabatic_electrons(self, like_species: str):
        """Add an adiabatic electron species following an existing species"""
        self.add_adiabatic_species(
            "electron", like_species, mass=electron_mass, charge=electron_charge
        )
```

Then in e.g. `GKInputGS2.get_local_species`, we'd have something like:

```python
if nspec == 1:
    local_species.add_adiabatic_electrons(local_species.names[0])
```

**Comment by bpatel2107** [comment]
Yes I think so, but there would be additional flags that would need to be set. But this looks like a good way to go

**Comment by ZedThree** [comment]
I'll make a PR with the basics, and we can discuss what else needs to be done there

**Comment by d7919** [comment]
Fixed by PR #412
